code/modules/projectiles/projectile/magic.dm

```diff
@@ -688,3 +690,276 @@
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	var/turf/X = get_turf(target)
+	explosion(X, -1, exp_light, 0, flame_range = exp_fire)
+	. = ..()
+
+/obj/item/projectile/magic/runic_toxin
+	name = "Runic Toxin"
+	icon_state = "syringeproj"
+	flag = "magic"
+	damage = 1
+	damage_type = BRUTE
+	nodamage = FALSE
+	eyeblur = 10
+
+/obj/item/projectile/magic/runic_toxin/on_hit(target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/X = target
+		X.reagents.add_reagent(/datum/reagent/toxin, 10)
+
+/obj/item/projectile/magic/runic_death
+	name = "Runic Death"
+	icon_state = "antimagic"
+	flag = "magic"
+	impact_effect_type = /obj/effect/temp_visual/dir_setting/bloodsplatter
+
+/obj/item/projectile/magic/runic_death/on_hit(mob/living/target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+	if(iszombie(target))
+		target.gib()
+	if(isskeleton(target))
+		target.gib()
+	if(isvampire(target))
+		target.adjustBruteLoss(40)
+
+
+/obj/item/projectile/magic/shotgun_slug
+	name = "Shotgun slug"
+	icon_state = "bullet"
+	damage = 10
+	damage_type = BRUTE
+	nodamage = FALSE
+	flag = "magic"
+
+/obj/item/projectile/magic/shotgun_slug/on_hit(target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+
+/obj/item/projectile/magic/incediary_slug
+	name = "Incendiary shotgun slug"
+	icon_state = "bullet"
+	damage = 5
+	damage_type = BURN
+	nodamage = FALSE
+	flag = "magic"
+
+
+/obj/item/projectile/magic/incediary_slug/on_hit(target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/X = target
+		X.fire_stacks += 1
+		X.IgniteMob()
+
+/obj/item/projectile/magic/runic_mutation
+	name = "Runic Mutation"
+	icon_state = "toxin"
+	flag = "magic"
+	irradiate = 12
+/obj/item/projectile/magic/runic_mutation/on_hit(target)
+	if(ismob(target))
+		var/mob/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			qdel(src)
+			return BULLET_ACT_BLOCK
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/X = target
+		X.randmuti()
+		if(prob(50))
+			X.easy_randmut(NEGATIVE)
```

Comment:
I'd say just make it always give a negative on hit, seems very weak if it can give minor stuff along with the chance of stuff like TK